### PR TITLE
Speedup mach startup

### PR DESCRIPTION
**Pull request by aidanhs**
...by using a 'marker file' to indicate whether we actually need to run pip.

Also a minor tweak for clarity.

Before (consistently):

```
$ time ./mach >/dev/null

real    0m0.666s
user    0m0.477s
sys     0m0.190s
```

After:

```
$ time ./mach >/dev/null # first run

real    0m0.665s
user    0m0.501s
sys     0m0.166s
$ time ./mach >/dev/null

real    0m0.121s
user    0m0.083s
sys     0m0.039s
```

[Review on Reviewable](https://reviewable.io/reviews/servo/servo/7632)

### CODE REVIEW

**Comment by aidanhs** [comment]
I've added a grace period of 10s to help avoid (unlikely) race conditions with a slow pip install as you checkout another branch/pull some new code.

**Comment by Ms2ger** [comment]
I'd like to see this submitted upstream: https://bugzilla.mozilla.org/enter_bug.cgi?product=Core&component=mach

**Comment by aidanhs** [comment]
Can you clarify what you want upstreaming?

After eventually finding where mach lives upstream (https://hg.mozilla.org/mozilla-central/file/default/mach, https://hg.mozilla.org/mozilla-central/file/default/build/mach_bootstrap.py), I couldn't see any invocations of `pip` and the replacement of `[0:0]` would need to happen in `mach_bootstrap.py` instead...but is that worth taking by itself?

Does the pip install happen somewhere else?

**Comment by aidanhs** [comment]
```
/css21_dev/html4/font-family-013.htm
------------------------------------
FAIL [Parent]
/css21_dev/html4/fonts-013.htm
------------------------------
```

**Comment by mbrubeck** [comment]
This is an issue with our test infrastructure, #7625. We'll sort this out soon.

**Comment by jdm** [comment]
@bors-servo: r-
Oh, I didn't notice that the original attempt to merge was a try.

**Comment by metajack** [comment]
@bors-servo r+

---

Reviewed 1 of 1 files at r1.
Review status: 1 of 2 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7632)

File: python/mach_bootstrap.py [hunk 1]
```diff
@@ -96,12 +96,21 @@ def _activate_virtualenv(topdir):
     # chain each of the requirements files into the same `pip install` call
     # and it will check for conflicts.
     requirements_paths = [
-        os.path.join(topdir, "python", "requirements.txt"),
```
Comment (frewsxcv) [inline python/mach_bootstrap.py]: Why was this removed?

Comment (frewsxcv) [inline python/mach_bootstrap.py]: Never mind. I see the logic was moved below in the `for..in`
